src/Castle.Windsor.Tests/ReferenceTracker.cs

```diff
@@ -0,0 +1,137 @@
+﻿// Copyright 2018 Castle Project - http://www.castleproject.org/
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace CastleTests
+{
+	using System;
+
+	using NUnit.Framework;
+
+	public static class ReferenceTracker
+	{
+		/// <summary>
+		/// <para>
+		/// Use a lambda method to evaluate an expression returning an object instance to track.
+		/// </para>
+		/// <para>
+		/// (If the expression is evaluated directly inside the test method, the runtime will keep it alive
+		/// until the test method returns if the runtime is in debug mode. ReSharper’s test runner runs in
```

Comment:
(This isn't so much a request to rephrase anything as it's just me being curious, so feel free to ignore.)

I was wondering about the accuracy of _"if the runtime is in debug mode"_. Is it actually the JIT (controlled via `[assembly: Debuggable]`, IIRC), or is it the C# compiler outputting different code, that causes the difference in behavior?

Reply:
It is both. The C# compiler performs IL optimization in release mode and not in debug mode, and the JIT performs optimizations unless `Debuggable` disables them.

With VSTest we have an example of running unoptimized IL in a process that has JIT optimizations enabled. This means that it's either all about the JIT operating in debug mode, or else it's a combination of the C# compiler generating unoptimized IL *and* the JIT operating in debug mode.

Reply:
Interesting! Thanks for checking up on this & letting me know. Quite happy to leave the comment as is.